src/main/java/gr/aueb/card/Tree.java

```diff
@@ -104,20 +106,61 @@ private void drawTrunk(Graphics g) {
      * @see gr.aueb.Tree the class description.
      */
     private void drawBody(Graphics g) {
-	/* Create the polygon (triangle) to draw */
-	Polygon p = new Polygon();
-	p.addPoint(dimensions.x + dimensions.width / 2, dimensions.y);
-	p.addPoint(dimensions.x,
-		(int) (dimensions.y + dimensions.height * bodyHeightFactor));
-	p.addPoint(dimensions.x + dimensions.width,
-		(int) (dimensions.y + dimensions.height * bodyHeightFactor));
-	/* Draw the body */
-	g.drawPolygon(p);
-
-	/* Fill it with green color */
-	Color c = g.getColor();
-	g.setColor(green);
-	g.fillPolygon(p);
-	g.setColor(c); // Revert paint color to default
+	    /* Create the polygon (triangle) to draw */
+	    Polygon p = new Polygon();
+	    p.addPoint(dimensions.x + dimensions.width / 2, dimensions.y);
+	    p.addPoint(dimensions.x,
+		    (int) (dimensions.y + dimensions.height * bodyHeightFactor));
+	    p.addPoint(dimensions.x + dimensions.width,
+		    (int) (dimensions.y + dimensions.height * bodyHeightFactor));
+	    /* Draw the body */
+	    g.drawPolygon(p);
+
+	    /* Fill it with green color */
+	    Color c = g.getColor();
+	    g.setColor(green);
+	    g.fillPolygon(p);
+	    g.setColor(c); // Revert paint color to default
+
+        /* Set Ornaments to the body. */
+        setTreeOrnaments(g);
+    }
+
+    /**
+     * Draws the ornaments of the tree.
+     * @param g The Graphics object on which we will paint
+     * @param x The Abscissa of the part of the body to draw the ornament
+     * @param y The Ordinate of the part of the body to draw the ornament
+     */
+    private void addTreeOrnament(Graphics g, int x, int y) {
+        /* Draw Tree Ornament. */
+        g.drawOval(x, y, 10, 10);
+        /* Set color to Red. */
+        g.setColor(red);
+        /* Fill Tree Ornament with color. */
+        g.fillOval(x, y, 10, 10);
+    }
+
+    /**
+     * Calls addTreeOrnament for specific locations on
+     * the tree body.
+     * @param g The Graphics object on which we will paint
+     */
+    private void setTreeOrnaments(Graphics g) {
+        /* Add ornament to down left. */
+        addTreeOrnament(g, dimensions.x -2,
+                (int) (dimensions.y + dimensions.height * bodyHeightFactor) -2);
```

Comment:
Consider pushing the repeated code `  (int) (dimensions.y + dimensions.height * bodyHeightFactor)` to the called function.